providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/grpc.py

```diff
@@ -105,81 +112,87 @@ def connect(self) -> None:
                 "Blocking init finished before data synced. Consider increasing startup deadline to avoid inconsistent evaluations."
             )
 
+    def monitor(self) -> None:
+        def state_change_callback(new_state: ChannelConnectivity) -> None:
```

Comment:
can you move the inner function to the class level? I always prefer not have inner functions, if possible. too many times I had to extract them later on, because of strange side effects.

Reply:
thank you, extracted into a "private" method